src/cli/repl/meta_cmd.rs

```diff
@@ -0,0 +1,711 @@
+            "Otherwise the full help for the command in the first argument is printed.",
+        ],
+        example: &[
+            "!(help)",
+            "!(help verify)",
+            "!(help \"load\")",
+        ],
+        run: |repl, cmd, args, _pwd_path| {
+            let first = repl.peek1(cmd, args)?;
+            match first.tag {
+                ExprTag::Str => {
+                    let name = repl.get_string(&first)?;
+                    Self::meta_help(&name);
+                }
+                ExprTag::Sym => {
+                    let sym = repl.get_symbol(&first)?;
+                    let name = sym.path().last().unwrap();
+                    Self::meta_help(name);
+                }
+                ExprTag::Nil => {
+                    use itertools::Itertools;
+                    println!("Available commands:");
+                    for (_, i) in MetaCmd::cmds().iter().sorted_by_key(|x| x.0) {
+                        println!("  {} - {}", i.name, i.summary);
+                    }
+                },
+                _ => bail!("The optional argument of `help` must be a string or symbol"),
+            }
+            return Ok(());
+        }
+    };
+
+    fn meta_help(cmd: &str) {
+        match MetaCmd::cmds().get(cmd) {
+            Some(i) => {
+                println!("{} - {}", i.name, i.summary);
+                for &e in i.description.iter() {
+                    println!("  {}", e);
+                }
+                println!("  Usage: {}", i.format);
+                if i.example.len() > 0 {
+                    println!("  Example:");
+                }
+                for &e in i.example.iter() {
+                    println!("    {}", e);
+                }
+            },
+            None => println!("unknown command {}", cmd),
+        }
+    }
+}
+
+// -------------------------------------------------------------------------- //
+// Meta Command Map
+
+impl MetaCmd<F> {
+    pub(super) fn cmds() -> std::collections::HashMap<&'static str, MetaCmd<F>> {
+        return std::collections::HashMap::from([
+            ("load", MetaCmd::LOAD),
+            ("def", MetaCmd::DEF),
+            ("defrec", MetaCmd::DEFREC),
+            ("assert", MetaCmd::ASSERT),
+            ("assert-eq", MetaCmd::ASSERT_EQ),
+            ("assert-emitted", MetaCmd::ASSERT_EMITTED),
+            ("assert-error", MetaCmd::ASSERT_ERROR),
+            ("commit", MetaCmd::COMMIT),
+            ("hide", MetaCmd::HIDE),
+            ("fetch", MetaCmd::FETCH),
+            ("open", MetaCmd::OPEN),
+            ("clear", MetaCmd::CLEAR),
+            ("set-env", MetaCmd::SET_ENV),
+            ("prove", MetaCmd::PROVE),
+            ("verify", MetaCmd::VERIFY),
+            ("defpackage", MetaCmd::DEFPACKAGE),
+            ("import", MetaCmd::IMPORT),
+            ("in-package", MetaCmd::IN_PACKAGE),
+            ("help", MetaCmd::HELP),
+        ]);
```

Comment:
Here it's better to iterate on a list of meta commands and include pairs of key/values where the keys are extracted from the commands `name` field to avoid repeating more hand-written strings.